src/web/midlw.rs

```diff
@@ -146,3 +144,307 @@ pub async fn auth(
+    });
+
+    #[tokio::test]
+
+    async fn test_authenticated_issuer_from_request_parts_success() {
+        *INIT;
+        // Generate keypair and JWT
+        let keypair = Keypair::generate().unwrap();
+        let public_key_pem = keypair
+            .verifying_key()
+            .to_public_key_pem(LineEnding::default())
+            .unwrap();
+        let private_key_pem = keypair.to_pkcs8_pem_bytes().unwrap();
+
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap()
+            .as_secs() as usize;
+        let issuer_id = "test-issuer-demo";
+        #[derive(serde::Serialize)]
+        struct Claims {
+            exp: usize,
+            iat: usize,
+        }
+        let claims = Claims {
+            exp: now + 3600,
+            iat: now,
+        };
+        let mut header = JwtHeader::new(jsonwebtoken::Algorithm::ES256);
+        header.kid = Some(issuer_id.to_string());
+        let token = encode(
+            &header,
+            &claims,
+            &EncodingKey::from_ec_pem(&private_key_pem).unwrap(),
+        )
+        .unwrap();
+
+        // Setup AppState with the issuer registered
+        let app_state = AppState::setup_test_with_credential(
+            issuer_id,
+            &public_key_pem,
+            jsonwebtoken::Algorithm::ES256,
+        )
+        .await;
+
+        // Build request parts with Authorization header
+        let mut headers = HeaderMap::new();
+        headers.insert(
+            header::AUTHORIZATION,
+            format!("Bearer {}", token).parse().unwrap(),
+        );
+        let mut parts = Request::builder()
+            .method(Method::GET)
+            .uri("/")
+            .body(Body::empty())
+            .unwrap()
+            .into_parts()
+            .0;
+        parts.headers = headers;
+
+        // Call the middleware
+        let result = AuthenticatedIssuer::from_request_parts(&mut parts, &app_state).await;
+        assert!(result.is_ok());
+        let authenticated_issuer = result.unwrap();
+        assert_eq!(authenticated_issuer.0, issuer_id);
+    }
+
+    #[tokio::test]
+
+    async fn test_missing_authorization_header() {
+        *INIT;
+        let app_state = crate::utils::state::setup().await;
+        let mut parts = Request::builder()
+            .method(Method::GET)
+            .uri("/")
+            .body(Body::empty())
+            .unwrap()
+            .into_parts()
+            .0;
+        // No headers set
+        let result = AuthenticatedIssuer::from_request_parts(&mut parts, &app_state).await;
+        assert!(result.is_err());
+        let resp = result.err().unwrap().into_response();
+        assert_eq!(resp.status(), StatusCode::UNAUTHORIZED);
+        // Check body
+        let body = futures::executor::block_on(async {
+            let bytes = to_bytes(resp.into_body(), 1024 * 1024).await.unwrap(); // Set limit to 1MB
+            String::from_utf8(bytes.to_vec()).unwrap()
+        });
```

Comment:
You're already in an async function. I wonder why you need a `block_on` directive. The same comment goes for all the subsequent occurrences of this syntax.
```suggestion
            let bytes = to_bytes(resp.into_body(), 1024 * 1024).await.unwrap(); // Set limit to 1MB
            let body = String::from_utf8(bytes.to_vec()).unwrap();
```

Reply:
OKay sir I will work on that.